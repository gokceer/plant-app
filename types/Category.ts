export interface CategoryImage {
  id: number;
  name: string;
  alternativeText: string | null;
  caption: string | null;
  width: number;
  height: number;
  formats: any;
  hash: string;
  ext: string;
  mime: string;
  size: number;
  url: string;
  previewUrl: string | null;
  provider: string;
  provider_metadata: any;
  createdAt: string;
  updatedAt: string;
}

export interface Category {
  id: number;
  name: string;
  createdAt: string;
  updatedAt: string;
  publishedAt: string;
  title: string;
  rank: number;
  image: CategoryImage;
}
