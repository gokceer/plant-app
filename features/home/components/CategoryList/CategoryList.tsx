import { Category } from "@/types/Category";
import React from "react";
import { FlatList, StyleSheet, View } from "react-native";
import CategoryBox from "../CategoryBox/CategoryBox";

const CategoryList = ({ categories }: { categories: Category[] }) => {
  return (
    <View>
      <FlatList
        data={categories}
        keyExtractor={(_, index) => index.toString()}
        renderItem={({ item }) => (
          <CategoryBox image={item.image} title={item.title} />
        )}
        numColumns={2}
        columnWrapperStyle={styles.row}
        scrollEnabled={false}
      />
    </View>
  );
};

export default CategoryList;

const styles = StyleSheet.create({
  row: {
    justifyContent: "space-between",
    marginBottom: 16,
    marginHorizontal: 24,
  },
  card: {
    backgroundColor: "#ddd",
    height: 100,
    flex: 1,
    marginRight: 10,
    borderRadius: 8,
  },
});
