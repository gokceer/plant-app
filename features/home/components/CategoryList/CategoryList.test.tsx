import { Category } from "@/types/Category";
import { render } from "@testing-library/react-native";
import React from "react";
import CategoryList from "./CategoryList";

describe("CategoryList Component", () => {
  const mockCategories: Category[] = [
    {
      id: 1,
      title: "Category 1",
      image: { url: "https://example.com/image1.jpg" },
    },
    {
      id: 2,
      title: "Category 2",
      image: { url: "https://example.com/image2.jpg" },
    },
    {
      id: 3,
      title: "Category 3",
      image: { url: "https://example.com/image3.jpg" },
    },
    {
      id: 4,
      title: "Category 4",
      image: { url: "https://example.com/image4.jpg" },
    },
  ];

  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<CategoryList categories={mockCategories} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("handles empty categories array", () => {
    const { queryByTestId } = render(<CategoryList categories={[]} />);
    const categoryBoxes = queryByTestId("categoryBox");

    expect(categoryBoxes).toBeNull();
  });
});
