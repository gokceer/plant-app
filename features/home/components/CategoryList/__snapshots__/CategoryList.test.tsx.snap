// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList Component renders correctly and matches snapshot 1`] = `
<View>
  <RCTScrollView
    data={
      [
        {
          "id": 1,
          "image": {
            "url": "https://example.com/image1.jpg",
          },
          "title": "Category 1",
        },
        {
          "id": 2,
          "image": {
            "url": "https://example.com/image2.jpg",
          },
          "title": "Category 2",
        },
        {
          "id": 3,
          "image": {
            "url": "https://example.com/image3.jpg",
          },
          "title": "Category 3",
        },
        {
          "id": 4,
          "image": {
            "url": "https://example.com/image4.jpg",
          },
          "title": "Category 4",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "justifyContent": "space-between",
                "marginBottom": 16,
                "marginHorizontal": 24,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "borderColor": "#29BB892E",
                  "borderRadius": 12,
                  "borderWidth": 0.5,
                  "marginRight": 10,
                  "overflow": "hidden",
                  "width": 346,
                },
                {
                  "height": 311.40000000000003,
                },
              ]
            }
            testID="categoryBox"
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  {
                    "uri": "https://example.com/image1.jpg",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <Text
                numberOfLines={0}
                style={
                  {
                    "color": "#13231B",
                    "fontFamily": "RubikMedium",
                    "fontSize": 16,
                    "padding": 16,
                    "width": "80%",
                  }
                }
              >
                Category 1
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "borderColor": "#29BB892E",
                  "borderRadius": 12,
                  "borderWidth": 0.5,
                  "marginRight": 10,
                  "overflow": "hidden",
                  "width": 346,
                },
                {
                  "height": 311.40000000000003,
                },
              ]
            }
            testID="categoryBox"
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  {
                    "uri": "https://example.com/image2.jpg",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <Text
                numberOfLines={0}
                style={
                  {
                    "color": "#13231B",
                    "fontFamily": "RubikMedium",
                    "fontSize": 16,
                    "padding": 16,
                    "width": "80%",
                  }
                }
              >
                Category 2
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "justifyContent": "space-between",
                "marginBottom": 16,
                "marginHorizontal": 24,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "borderColor": "#29BB892E",
                  "borderRadius": 12,
                  "borderWidth": 0.5,
                  "marginRight": 10,
                  "overflow": "hidden",
                  "width": 346,
                },
                {
                  "height": 311.40000000000003,
                },
              ]
            }
            testID="categoryBox"
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  {
                    "uri": "https://example.com/image3.jpg",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <Text
                numberOfLines={0}
                style={
                  {
                    "color": "#13231B",
                    "fontFamily": "RubikMedium",
                    "fontSize": 16,
                    "padding": 16,
                    "width": "80%",
                  }
                }
              >
                Category 3
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "borderColor": "#29BB892E",
                  "borderRadius": 12,
                  "borderWidth": 0.5,
                  "marginRight": 10,
                  "overflow": "hidden",
                  "width": 346,
                },
                {
                  "height": 311.40000000000003,
                },
              ]
            }
            testID="categoryBox"
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  {
                    "uri": "https://example.com/image4.jpg",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <Text
                numberOfLines={0}
                style={
                  {
                    "color": "#13231B",
                    "fontFamily": "RubikMedium",
                    "fontSize": 16,
                    "padding": 16,
                    "width": "80%",
                  }
                }
              >
                Category 4
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
